lib/assessment.js

```diff
@@ -13,14 +19,25 @@ var min = function min() {
 // and returns the sum of that set
 // assume that all the argument are numbers
 var sum = function sum() {
-
+  var total = 0;
+  for (var i = 0; i < arguments.length; i++) {
+    total = total + arguments[i];
+  }
+  return total;
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
+var defaultArray = [];
 var newArray = function newArray(length, defaultValue) {
-
+var i = 0;
+while (length > 0) {
+  defaultArray[i] = defaultValue;
+  length = length - 1;
+  i = i + 1;
+}
+return defaultArray;
 };
```

Comment:
So, this function would have been great if the array it returns were created inside the function. Since it's created outside the function, every time this function is run, all it is overwrite the same array. That means *the* (as in it returns the same specific array each time) array returned by this function is **extremely unsafe!** It can be overwritten unexpectedly without notice!